### PR TITLE
Release branch 2.15.4

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
Draft, because I had to do some tests for #7589 and have pending code. This could be released once the new astorid version is released and upgraded.

### CODE REVIEW

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Could you split the backporting and releasing? That makes reviewing these a lot simpler.

**Comment by Pierre-Sassoulas** [comment]
Sure, I forgot about that.

**Comment by Pierre-Sassoulas** [comment]
Blocked by #7591

**Comment by Pierre-Sassoulas** [comment]
Blocked by a new release of astroid and un upgrade in pylint.

File: doc/user_guide/checkers/features.rst [hunk 1]
```diff
@@ -876,6 +876,9 @@ Refactoring checker Messages
   Emitted when accessing only the first or last element of str.split(). The
   first and last element can be accessed by using str.split(sep, maxsplit=1)[0]
   or str.rsplit(sep, maxsplit=1)[-1] instead.
+:consider-using-augmented-assign (C0210): *Use '%s' to do an augmented assign directly*
```
Comment (DanielNoord) [inline doc/user_guide/checkers/features.rst:879]: Isn't it weird this change shows up in a patch release? Did we forget to update the docs last release?

File: doc/whatsnew/2/2.15/index.rst [hunk 2]
```diff
@@ -29,6 +29,54 @@ Marc Byrne became a maintainer, welcome to the team !
...
+  Closes #7529 (`#7529 <https://github.com/PyCQA/pylint/issues/7529>`_)
+
+
+Other Bug Fixes
+---------------
+
+- Fix bug in detecting ``unused-variable`` when iterating on variable.
+
+  Closes #3044 (`#3044 <https://github.com/PyCQA/pylint/issues/3044>`_)
+- Fixed handling of ``--`` as separator between positional arguments and flags.
```
Comment (DanielNoord) [inline doc/whatsnew/2/2.15/index.rst:52]: Can we backport the fix to `towncrier`?

File: doc/user_guide/checkers/features.rst (hunk 1 above)
Comment (Pierre-Sassoulas) [reply doc/user_guide/checkers/features.rst:879]: It's cherry-picked from main and should stay on main, let me fix that.

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**